File: scripts/generate_thumbnails.js
#!/usr/bin/env node

// Simple script to generate art style thumbnails using the OpenAI API
const fs = require('fs');
const path = require('path');
const https = require('https');

// OpenAI API key - replace with your key if not using environment variable
const OPENAI_API_KEY = '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';

// Directory to save the generated thumbnails
const OUTPUT_DIR = path.join(__dirname, '../public/assets/style-thumbnails');

// Create output directory if it doesn't exist
if (!fs.existsSync(OUTPUT_DIR)) {
  fs.mkdirSync(OUTPUT_DIR, { recursive: true });
  console.log(`Created output directory: ${OUTPUT_DIR}`);
}

// Art styles to generate
const artStyles = [
  {
    id: 'watercolor_storybook',
    prompt: 'Create a soft watercolor children\'s book illustration of a teddy bear. Gentle pastel colors, light brushstrokes, warm atmosphere, simple natural background.'
  },
  {
    id: 'cartoon_semi_realistic',
    prompt: 'Create a cartoon-style teddy bear character. Bold outlines, colorful, expressive, children\'s book energy.'
  },
  {
    id: 'storybook_fantasy',
    prompt: 'Create a richly detailed fantasy storybook teddy bear. Magical elements, lush textures, and glowing ambiance.'
  }
];

/**
 * Generate an image using OpenAI API
 */
function generateImage(prompt, styleId) {
  return new Promise((resolve, reject) => {
    console.log(`Generating image for style: ${styleId}`);
    
    const data = JSON.stringify({
      model: "gpt-image-1",
      prompt: prompt,
      n: 1,
      size: "1024x1024",
      quality: "high",
      response_format: "b64_json"
    });
    
    const options = {
      hostname: 'api.openai.com',
      path: '/v1/images/generations',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${OPENAI_API_KEY}`,
        'Content-Length': data.length
      }
    };
    
    const req = https.request(options, (res) => {
      let responseData = '';
      
      res.on('data', (chunk) => {
        responseData += chunk;
      });
      
      res.on('end', () => {
        if (res.statusCode === 200) {
          try {
            const parsedData = JSON.parse(responseData);
            if (parsedData.data && parsedData.data.length > 0) {
              const imageData = parsedData.data[0].b64_json;
              const buffer = Buffer.from(imageData, 'base64');
              
              const filePath = path.join(OUTPUT_DIR, `${styleId}.png`);
              fs.writeFileSync(filePath, buffer);
              
              console.log(`Successfully saved image for style: ${styleId} to ${filePath}`);
              resolve(filePath);
            } else {
              console.error(`No image data in response for style: ${styleId}`);
              reject(new Error('No image data in response'));
            }
          } catch (error) {
            console.error(`Error parsing response for style ${styleId}:`, error);
            reject(error);
          }
        } else {
          console.error(`Error response for style ${styleId}: ${res.statusCode} ${res.statusMessage}`);
          console.error(responseData);
          reject(new Error(`API returned status code ${res.statusCode}`));
        }
      });
    });
    
    req.on('error', (error) => {
      console.error(`Error generating image for style ${styleId}:`, error);
      reject(error);
    });
    
    req.write(data);
    req.end();
  });
}

/**
 * Generate thumbnails for all art styles
 */
async function generateThumbnails() {
  console.log(`Starting generation of ${artStyles.length} art style thumbnails...`);
  
  // Process styles sequentially to avoid rate limiting
  for (const style of artStyles) {
    try {
      await generateImage(style.prompt, style.id);
      // Add a delay between requests to avoid rate limiting
      await new Promise(resolve => setTimeout(resolve, 2000));
    } catch (error) {
      console.error(`Failed to generate thumbnail for ${style.id}:`, error);
    }
  }
  
  console.log('Thumbnail generation complete!');
  console.log(`Thumbnails saved to: ${OUTPUT_DIR}`);
}

// Start the generation process
generateThumbnails();

File: scripts/generate_thumbnails.py
#!/usr/bin/env python3
"""
Script to generate art style thumbnails using the OpenAI API
"""

import os
import base64
import time
from openai import OpenAI

# OpenAI API key - replace with your actual key when running
API_KEY = "YOUR_OPENAI_API_KEY"

# Output directory
OUTPUT_DIR = "public/assets/style-thumbnails"

# Create output directory if it doesn't exist
os.makedirs(OUTPUT_DIR, exist_ok=True)
print(f"Output directory: {OUTPUT_DIR}")

# Initialize OpenAI client
client = OpenAI(api_key=API_KEY)

# Art styles to generate
art_styles = [
    {
        "id": "watercolor_storybook",
        "name": "Watercolor Storybook",
        "prompt": "Create a soft watercolor children's book illustration of a teddy bear. Gentle pastel colors, light brushstrokes, warm atmosphere, simple natural background."
    },
    {
        "id": "cartoon_semi_realistic",
        "name": "Cartoon Character (Semi-Realistic)",
        "prompt": "Create a cartoon-style teddy bear character. Bold outlines, colorful, expressive, children's book energy."
    },
    {
        "id": "storybook_fantasy",
        "name": "Storybook Fantasy Style (Detailed)",
        "prompt": "Create a richly detailed fantasy storybook teddy bear. Magical elements, lush textures, and glowing ambiance."
    }
]

def generate_image(style_id, prompt):
    """Generate an image for a style using the OpenAI API"""
    output_file = os.path.join(OUTPUT_DIR, f"{style_id}.png")

    print(f"Generating image for style: {style_id}")
    print(f"Prompt: {prompt}")

    try:
        # Make the API request
        response = client.images.generate(
            model="gpt-image-1",
            prompt=prompt,
            n=1,
            size="1024x1024",
            quality="high",
            response_format="b64_json"
        )

        # Get the image data
        image_data = response.data[0].b64_json
        image_bytes = base64.b64decode(image_data)

        # Save the image
        with open(output_file, "wb") as f:
            f.write(image_bytes)

        print(f"✅ Successfully generated image for {style_id}")
        print(f"   Saved to: {output_file}")
        return True

    except Exception as e:
        print(f"❌ Failed to generate image for {style_id}")
        print(f"   Error: {str(e)}")
        return False

def main():
    """Main function to generate all thumbnails"""
    print(f"Starting generation of {len(art_styles)} art style thumbnails...")

    success_count = 0

    # Process styles sequentially
    for style in art_styles:
        if generate_image(style["id"], style["prompt"]):
            success_count += 1

        # Add a delay to avoid rate limiting
        time.sleep(2)

    print(f"Generation complete! Successfully generated {success_count}/{len(art_styles)} thumbnails.")
    print(f"Check the {OUTPUT_DIR} directory for the generated images.")

if __name__ == "__main__":
    main()
